### PR TITLE
noncliff: fixes `expect`

### PR DESCRIPTION
**Pull request by Fe-r-oz**
All the tests in `noncliffordquantumoptics.jl `should pass now.

If you want to submit an unfinished piece of work in order to get comments and discuss, please mark the pull request as a draft and ping the repository maintainer.

**Please address only one topic or issue per pull request! Many small PRs are much easier to review and merge than one large PR.**

Before merging, all changes and new functionality should be marked in the CHANGELOG file, but feel free to just leave your CHANGELOG notes in the PR description, to avoid merge conflicts with other requests modifying that file. The maintainer will add these CHANGELOG notes for you if you do so.

Before considering your pull request ready for review and merging make sure that all of the following are completed (please keep the clecklist as part of your PR):

- [x] The code is properly formatted and commented.
- [x] Substantial new functionality is documented within the docs.
- [x] All new functionality is tested.
- [x] All of the automated tests on github pass.

If possible, keep your git history not too wild (rebase and squash commits, keep commits small and semantically separated) so that review is easier.

### CODE REVIEW

**Comment by Fe-r-oz** [comment]
All the tests passed. This fixes `expect` from #259.  `CI-nightly` errors not related to this pr. I request review as no further commits will be pushed.

**Comment by Fe-r-oz** [comment]
@Krastanov, pinging for review. Thank you.

File: src/nonclifford.jl [hunk 1]
```diff
@@ -98,13 +98,13 @@ end
...
 function expect(p::PauliOperator, s::GeneralizedStabilizer) # TODO optimize
-    e = zero(_dictvaltype(s.destabweights))
+    χ′ = zero(_dictvaltype(s.destabweights))
     phase, b, c = rowdecompose(p, s.stab)
     for ((dᵢ,dⱼ), χ) in s.destabweights
         _allthreesumtozero(dᵢ,dⱼ,b) || continue
-        e += χ * (-1)^(dᵢ'*c)
+        χ′ += χ * (-1)^(dᵢ'*c)
     end
-    return (-1)^(phase÷2) * e
```
Comment (Krastanov) [inline src/nonclifford.jl:107]: oh, neat! That explains the poor results previously! Good find!

**Comment by Krastanov** [comment]
thanks, fixing this was pretty important! Neat to see how minimal the fix was

**Pull request merged**